src/main/java/net/neoforged/moddevgradle/internal/CreateLaunchScriptTask.java

```diff
@@ -0,0 +1,215 @@
+package net.neoforged.moddevgradle.internal;
+
+import net.neoforged.moddevgradle.dsl.RunModel;
+import net.neoforged.moddevgradle.internal.utils.ExtensionUtils;
+import net.neoforged.moddevgradle.internal.utils.FileUtils;
+import net.neoforged.moddevgradle.internal.utils.OperatingSystem;
+import net.neoforged.moddevgradle.internal.utils.StringUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Classpath;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.jvm.toolchain.JavaLauncher;
+import org.gradle.jvm.toolchain.JavaToolchainService;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Writes standalone start scripts to launch the game.
+ * These scripts are useful for launching the game outside Gradle or the IDE, for example, from tools
+ * like Renderdoc, NVidia NSight, Linux perf record and similar.
+ */
+abstract class CreateLaunchScriptTask extends DefaultTask {
+    @Input
+    abstract Property<String> getWorkingDirectory();
+
+    @InputFile
+    abstract Property<String> getVmArgsFile();
+
+    /**
+     * Set to {@link PrepareRun#getProgramArgsFile()}
+     */
+    @InputFile
+    abstract Property<String> getProgramArgsFile();
+
+    /**
+     * This argument file is only used by the launch shell-scripts.
+     */
+    @OutputFile
+    @Optional
+    abstract RegularFileProperty getClasspathArgsFile();
+
+    /**
+     * A platform-specific script to launch this run configuration with directly from outside the IDE / Gradle.
+     */
+    @OutputFile
+    @Optional
+    abstract RegularFileProperty getLaunchScript();
+
+    /**
+     * Set to the desired Java runtime classpath.
+     */
+    @Classpath
+    @InputFiles
+    abstract ConfigurableFileCollection getRuntimeClasspath();
+
+    @Input
+    abstract Property<ModFoldersProvider> getModFolders();
+
+    /**
+     * @see RunModel#getEnvironment()
+     */
+    @Input
+    abstract MapProperty<String, String> getEnvironment();
+
+    /**
+     * Launcher for the java version used in the launch shell-scripts.
+     */
+    @Internal
```

Comment:
Shouldn't this be `@Input`? (And be fixed in the other tasks that use it as `@Internal`?)